lib/matplotlib/axes/_axes.py

```diff
@@ -7629,6 +7629,8 @@ def specgram(self, x, NFFT=None, Fs=None, Fc=None, detrend=None,
             Fc = 0  # same default as in mlab._spectral_helper()
         if noverlap is None:
             noverlap = 128  # same default as in mlab.specgram()
+        if Fs is None:
```

Comment:
As discussed in https://github.com/matplotlib/matplotlib/pull/9221#issuecomment-331897539, I think that `Fs` should remain None, and the underlying function made to do the right thing with `None`.  This function should not be calculating frequencies, but should just take them from the frequency output of the spectral helper.  

Sorry to be a pain, but I'm going to block on this.  If you need more details to modify the PR, I will try and take a look later today.

Reply:
Then I'll try to change the PR so as to modify the docstring so that it matches the fact that the default value is None. I will try to do this asap.

Reply:
Well, I think there is a problem with the code as well, but I don't think setting `Fs=2` is the solution.  We should be making to `Fs` from the `freqs` array where it is needed in the code (`Fs=freqs[-1] * 2`)